lightning/src/ln/channel.rs

```diff
@@ -6103,8 +6128,22 @@ impl<'a, Signer: Sign, K: Deref> ReadableArgs<(&'a K, u32)> for Channel<Signer>
 			(17, announcement_sigs_state, option),
 			(19, latest_inbound_scid_alias, option),
 			(21, outbound_scid_alias, option),
+			(23, outbound_htlc_maximum_msat, option),
 		});
 
+		// Construct the `CounterpartyForwardingInfo` after the TLV Optional variant for
+		// `outbound_htlc_maximum_msat` has been read.
+		let counterparty_forwarding_info = match has_forwarding_info {
+			false => None,
```

Comment:
I dont think we can do this - if we're reading an object serialized by an older version of LDK this would imply we jump back to the channel being uninitialized and we'll panic on some unwraps above - we need to initialize the fields to some default.